File: erp-shop/src/main/java/com/own/erp/shop/entity/PullLog.java
package com.own.erp.shop.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * @author : chenyi
 * @Date : 2026/9/3
 * @Description : 平台拉取日志(增量游标依据:取成功记录的window_end,左叠5分钟)(pull_log)
 */
@Data
@TableName("pull_log")
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PullLog {

    /** 主键 */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 店铺ID(shop.id) */
    private Long shopId;

    /** ORDER/PRODUCT/REFUND */
    private String dataType;

    /** 拉取窗口起点(含) */
    private LocalDateTime windowStart;

    /** 拉取窗口终点(含);游标=最近成功记录的window_end */
    private LocalDateTime windowEnd;

    /** 本次拉取条数 */
    private Integer pulledCount;

    /** 1成功0失败 */
    private Integer success;

    /** 失败原因 */
    private String errorMsg;

    /** 本次拉取耗时(毫秒),观测慢店铺/慢接口 */
    private Integer durationMs;

    /** 触发方式:JOB定时/MANUAL手动 */
    private String pullWay;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;
}

File: erp-web/src/api/interface/shop/pull-log.ts
/**
 * 本文件由 pnpm gen:page 生成(spec 行式拍板 + tools/openapi.json 快照)
 * 默认存在即跳过:人工改动不会被 --force 之外的任何方式覆盖;重新生成前先 diff 人工改动
 * 框架代码禁手改;业务槽位一律 TODO(编号),编号已登记 TODO.md
 */

/** 拉单日志接口类型(gen:page 从 openapi 快照生成;对齐后端契约,禁手抄字段) */

/** 拉单日志实体(分页行 / 详情) */
export interface PullLogResponse {
  /** id */
  id: number
  /** 店铺ID */
  shopId: number
  /** 数据类型 */
  dataType: string
  /** 窗口起点 */
  windowStart: string
  /** 窗口终点 */
  windowEnd: string
  /** 拉取条数 */
  pulledCount: number
  /** 结果 */
  success: number
  /** 失败原因 */
  errorMsg: string
  /** 耗时(ms) */
  durationMs: number
  /** 触发方式 */
  pullWay: string
  /** 创建时间 */
  createdAt: string
  /** updatedAt */
  updatedAt: string
}

/** 拉单日志分页查询入参(不含分页参数) */
export interface PullLogQuery {
  /** 店铺ID */
  shopId?: number
  /** 数据类型 */
  dataType?: string
  /** 结果 */
  success?: number
}
